### ISSUE TITLE
doubt about model test

### ISSUE THREAD

**Issue by huabinhuang1994** [amdegroot/ssd.pytorch#204]
during the test , sevaral lines of code in the forward() function of class detection:
for i in range(num):
            decoded_boxes = decode(loc_data[i], prior_data, self.variance)
            # For each class, perform nms
            conf_scores = conf_preds[i].clone()
            for cl in range(1, self.num_classes):
                c_mask = conf_scores[cl].gt(self.conf_thresh)
                scores = conf_scores[cl][c_mask]
                if scores.dim() == 0:
                    continue
                l_mask = c_mask.unsqueeze(1).expand_as(decoded_boxes)
                boxes = decoded_boxes[l_mask].view(-1, 4)
                # idx of highest scoring and non-overlapping boxes per class
                ids, count = nms(boxes, scores, self.nms_thresh, self.top_k)
                output[i, cl, :count] = \
                    torch.cat((scores[ids[:count]].unsqueeze(1),
                               boxes[ids[:count]]), 1)
        flt = output.contiguous().view(num, -1, 5)
        _, idx = flt[:, :, 0].sort(1, descending=True)
        _, rank = idx.sort(1)
        flt[(rank < self.top_k).unsqueeze(-1).expand_as(flt)].fill_(0)
        return output 
we know that each box can output 21 classification score for the VOC Dataset , for one box ,if I get the it's softmax score like [0,0.02,0.02,0.8,...], then acording to this code , class 1,2,3 will all include this box. isn't this a bug? or I just ignore something.

**Comment by huabinhuang1994** [comment]
my fault ,another threshold is in the test function()
